kore/src/Kore/Strategies/Goal.hs

```diff
@@ -622,6 +623,47 @@ data TransitionRuleTemplate monad goal =
         -> Strategy.TransitionT (Rule goal) monad (ProofState goal goal)
     }
 
+data InfoReachability goal 
+    = InfoSimplify !goal
```

Comment:
I called this `InfoSimplify` instead of `Simplify` because otherwise the uses of `ProofState.Simplify` were ambiguous.
(same for the other 3 constructors)